### PR TITLE
body_exists for requests without body should be False

### PR DESCRIPTION
**Pull request by hh-h**
## What do these changes do?

This PR tries to implement [this logic](https://github.com/aio-libs/aiohttp/blob/4d03dbb32338cfda1234a3ba8628cf4b10e201d5/aiohttp/http_parser.py#L310) for `_http_parser` 

## Are there changes in behavior for the user?

No

## Related issue number

#4528 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

File: aiohttp/_http_parser.pyx [hunk 1]
```diff
@@ -427,8 +431,12 @@ cdef class HttpParser:
                 headers, raw_headers, should_close, encoding,
                 upgrade, chunked)
 
-        if (self._cparser.content_length > 0 or chunked or
-                self._cparser.method == 5):  # CONNECT: 5
+        if (ULLONG_MAX > self._cparser.content_length > 0 or chunked or
```
Comment (hh-h) [inline aiohttp/_http_parser.pyx:434]: https://github.com/nodejs/http-parser/blob/77310eeb839c4251c07184a5db8885a572a08352/http_parser.c#L1803

**Comment by hh-h** [comment]
Added CHANGES file

**Comment by asvetlov** [comment]
Please sync with the lastest master to get rid of unittest error

**Comment by hh-h** [comment]
Done, green 💚

**Comment by asvetlov** [comment]
Thanks!

**Pull request merged**